frame/support/src/lib.rs

```diff
@@ -28,6 +28,7 @@
 // of this crate.
 
 #![cfg_attr(not(feature = "std"), no_std)]
+#![allow(deprecated)]
```

Comment:
We should not do this. Then let's put this everywhere it is needed, but we can not allow this in the entire crate.